model_compression_toolkit/core/keras/hessian/weights_trace_hessian_calculator_keras.py

```diff
@@ -50,5 +53,112 @@ def __init__(self,
                                                                  trace_hessian_request=trace_hessian_request,
                                                                  num_iterations_for_approximation=num_iterations_for_approximation)
 
-    def compute(self):
-        Logger.error(f"Hessian trace approx w.r.t weights is not supported for now")
\ No newline at end of file
+    def compute(self) -> np.ndarray:
+        """
+        Compute the Hessian-based scores for w.r.t target node's weights.
+        Currently, supported nodes are [Conv2D, Dense, Conv2DTranspose, DepthwiseConv2D].
+        The computed scores are returned in a numpy array. The shape of the result differs
+        according to the requested granularity. If for example the node is Conv2D with a kernel
+        shape of (3, 3, 3, 2) (namely, 3 input channels, 2 output channels and kernel size of 3x3)
+        and the required granularity is HessianInfoGranularity.PER_TENSOR the result shape will be (1,),
+        for HessianInfoGranularity.PER_OUTPUT_CHANNEL the shape will be (2,) and for
+        HessianInfoGranularity.PER_ELEMENT a shape of (3, 3, 3, 2).
+
+        Returns:  The computed scores as numpy ndarray.
+
+        """
+        # Check if the target node's layer type is supported
+        if self.hessian_request.target_node.layer_class not in [Conv2D, Dense, Conv2DTranspose, DepthwiseConv2D]:
+            Logger.error(
+                f"{self.hessian_request.target_node.type} is not supported for Hessian info w.r.t weights.")
+
+        # Construct the Keras float model for inference
+        model, _ = FloatKerasModelBuilder(graph=self.graph).build_model()
+        # TODO: what to do in reuse
+
+        # Get the weight attributes for the target node type
+        weight_attributes = DEFAULT_KERAS_INFO.get_kernel_op_attributes(self.hessian_request.target_node.type)
+        assert len(weight_attributes) == 1
+
+        # Get the weight tensor for the target node
+        weight_tensor = getattr(model.get_layer(self.hessian_request.target_node.name), weight_attributes[0])
+
+        # Get the output channel index (needed for HessianInfoGranularity.PER_OUTPUT_CHANNEL case)
+        output_channel_axis, _ = DEFAULT_KERAS_INFO.kernel_channels_mapping.get(
+            self.hessian_request.target_node.type)
+
+        # Initiate a gradient tape for automatic differentiation
+        with tf.GradientTape(persistent=True) as tape:
+            # Perform a forward pass (inference) to get the output, while watching
+            # the input tensor for gradient computation
+            tape.watch(self.input_images)
```

Comment:
I know this was already done in previous PR but It is not smooth when the input is called input_images. (what if it is not images). input_tensors is more accurate.

Reply:
Since it's in other existing parts (like in w.r.t activations), I'll rename it in all those parts in a separate PR.